RadixWallet/Clients/AccountPortfoliosClient/AccountPortfoliosClient+State.swift

```diff
@@ -0,0 +1,285 @@
+			setOrUpdateAccountPortfolios(existingPortfolios)
+		}
+	}
+
+	func applyTokenPrices(to portfolios: inout [AccountPortfoliosClient.AccountPortfolio]) {
+		portfolios.mutateAll(applyTokenPrices)
+	}
+
+	func applyTokenPrices(to portfolio: inout AccountPortfoliosClient.AccountPortfolio) {
+		portfolio.updateFiatWorth(calculateWorth(gateway))
+	}
+
+	func applyCurrencyVisibility(to portfolio: inout AccountPortfoliosClient.AccountPortfolio) {
+		portfolio.isCurrencyAmountVisible = isCurrencyAmountVisible
+		portfolio.updateFiatWorth(value: isCurrencyAmountVisible, to: \.isVisible)
+	}
+
+	func applyCurrencyVisibility(to portfolios: inout [AccountPortfoliosClient.AccountPortfolio]) {
+		portfolios.mutateAll(applyCurrencyVisibility)
+	}
+
+	func applyFiatCurrency(to portfolio: inout AccountPortfoliosClient.AccountPortfolio) {
+		portfolio.fiatCurrency = self.selectedCurrency
+		portfolio.updateFiatWorth(value: selectedCurrency, to: \.currency)
+	}
+
+	func applyFiatCurrency(to portfolios: inout [AccountPortfoliosClient.AccountPortfolio]) {
+		portfolios.mutateAll(applyFiatCurrency)
+	}
+}
+
+// MARK: - Stake and Pool details handling
+extension AccountPortfoliosClient.State {
+	func calculateWorth(_ gateway: Radix.Gateway) -> (ResourceAddress, ResourceAmount) -> FiatWorth? {
+		{ resourceAddress, amount in
+			let price = {
+				#if DEBUG
+				if gateway != .mainnet {
+					if resourceAddress == .mainnetXRDAddress {
+						return self.tokenPrices[resourceAddress]
+					} else {
+						return self.tokenPrices.values.randomElement()
+					}
+				} else {
+					return self.tokenPrices[resourceAddress]
+				}
+				#else
+				return self.tokenPrices[resourceAddress]
+				#endif
+			}()
+
+			return price.map {
+				.init(
+					isVisible: self.isCurrencyAmountVisible,
+					worth: .known($0 * amount.nominalAmount),
+					currency: self.selectedCurrency
+				)
+			}
+		}
+	}
+
+	func set(poolDetails: Loadable<[OnLedgerEntitiesClient.OwnedResourcePoolDetails]>, forAccount address: AccountAddress) {
+		guard var portfolio = portfoliosSubject.value.wrappedValue?[address] else {
+			return
+		}
+
+		portfolio.poolUnitDetails = poolDetails.map { details in
+			var details = details
+			details.updateFiatWorth(calculateWorth(gateway))
+			return details
+		}
+		setOrUpdateAccountPortfolio(portfolio)
+	}
+
+	func set(stakeUnitDetails: Loadable<IdentifiedArrayOf<OnLedgerEntitiesClient.OwnedStakeDetails>>, forAccount address: AccountAddress) {
+		guard var portfolio = portfoliosSubject.value.wrappedValue?[address] else {
+			return
+		}
+		portfolio.stakeUnitDetails = stakeUnitDetails.map { details in
+			var details = details
+			details.updateFiatWorth(calculateWorth(gateway))
+			return details
+		}
+		setOrUpdateAccountPortfolio(portfolio)
+	}
+}
+
+// MARK: Fiat Worth changes
+private extension AccountPortfoliosClient.AccountPortfolio {
+	mutating func updateFiatWorth<T>(value: T, to keyPath: WritableKeyPath<FiatWorth, T>) {
```

Comment:
Updates the fiatWorth field by keypath, useful for updated isCurrencyAmountVisible or selectedFiatCurrency.